ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -716,6 +757,13 @@ void open(CompactionInfo ci) throws TException {
       }
       this.txnId = msc.openTxn(ci.runAs, TxnType.COMPACTION);
       status = TxnStatus.OPEN;
+
+      LockRequest lockRequest = createLockRequest(ci, txnId);
+      LockResponse res = msc.lock(lockRequest);
```

Comment:
I might have missed it but do we unlock this as well?

Reply:
Or does this happen automatically on commit/abort?

Reply:
automatically on commit/abort